x-pack/plugin/security/src/main/java/org/elasticsearch/xpack/security/action/apikey/TransportQueryApiKeyAction.java

```diff
@@ -68,10 +80,18 @@ protected void doExecute(Task task, QueryApiKeyRequest request, ActionListener<Q
 
         final ApiKeyBoolQueryBuilder apiKeyBoolQueryBuilder = ApiKeyBoolQueryBuilder.build(
             request.getQueryBuilder(),
-            request.isFilterForCurrentUser() ? authentication : null
+            filteringAuthentication
         );
         searchSourceBuilder.query(apiKeyBoolQueryBuilder);
 
+        AggregatorFactories.Builder aggsBuilder = request.getAggsBuilder();
+        // this modifies the aggsBuilder in-place
+        ApiKeyAggregationsBuilder.verifyRequested(aggsBuilder, filteringAuthentication);
+        searchSourceBuilder.aggregationsBuilder(aggsBuilder);
```

Comment:
The above verifies & modifies the requested aggregation to:
 * prohib scripts that can directly access the API Key docs
 * rewrite the field names (because the query & agg use different field names from the document ones, in order to reflect the terminology that we use in the returned API Keys)
 * rewrite queries to scope them to API Key docs only